examples/i18n/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+// Some Clippy linter rules are ignored for the sake of simplicity.
+#![allow(clippy::needless_pass_by_value, clippy::trivially_copy_pass_by_ref)]
+
+use seed::{prelude::*, *};
+
+use fluent::{FluentArgs, FluentBundle, FluentResource, FluentValue};
+use unic_langid::LanguageIdentifier;
+
+use enum_iterator::IntoEnumIterator;
+use std::borrow::Borrow;
+
+mod resource;
+
+// ------ ------
+//     Init
+// ------ ------
+
+fn init(_: Url, _: &mut impl Orders<Msg>) -> Model {
+    let mut model = Model::default();
+    change_lang(&mut model);
```

Comment:
Rust supports safe mutability but use it only when best practices for immutable functions and testability make the code too unreadable or slow.
I would rather see something like:

```rust
const DEFAULT_LANG: Lang = Lang::en_US;
...
fn init(_: Url, _: &mut impl Orders<Msg>) -> Model {
    let i18n = I18n::new(DEFAULT_LANG);
    Model { i18n }
}
```
or 
```rust
fn init(_: Url, _: &mut impl Orders<Msg>) -> Model {
    Model { 
        i18n: I18n::new(DEFAULT_LANG),
    }
}
```
where
```rust

struct Model {
    i18n: I18n
}

// ------ I18n ------

struct I18n {
    lang: Lang,
    resource: ...
}
```
=> No more implicit (and error-prone) behavior and you can't forget to call `change_lang`. Also writing unit tests for `change_lang` would be pain once your `Model` is big enough.

Reply:
Good lesson. I got rid of stand alone function "change_lang" by implementing a function to set lang within I18n. Still I struggle with I18n::new(). Please check again my implementation. Thanks.